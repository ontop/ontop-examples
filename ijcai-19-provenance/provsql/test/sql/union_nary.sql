\set ECHO none
SET search_path TO public,provsql;

CREATE TABLE nested_union_result AS
SELECT *,formula(provenance(),'personnel_name') FROM (
  SELECT city FROM personnel WHERE classification != 'unclassified'
  UNION
  SELECT city FROM personnel WHERE position != 'Janitor'
  UNION
  SELECT city FROM personnel WHERE name LIKE '%n'
  UNION
  SELECT city FROM personnel WHERE position LIKE '%agent'
) t;

SELECT remove_provenance('nested_union_result');
SELECT * FROM nested_union_result ORDER BY city;
DROP TABLE nested_union_result;
