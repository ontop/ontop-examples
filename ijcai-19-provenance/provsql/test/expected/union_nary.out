\set ECHO none
 remove_provenance 
-------------------
 
(1 row)

   city   |                                  formula                                  
----------+---------------------------------------------------------------------------
 Berlin   | (Ellen ⊕ Susan ⊕ Ellen ⊕ Susan ⊕ Ellen ⊕ Susan ⊕ Ellen)
 New York | (Paul ⊕ John ⊕ John)
 Paris    | (Dave ⊕ Magdalen ⊕ Nancy ⊕ Dave ⊕ Magdalen ⊕ Nancy ⊕ Magdalen ⊕ Magdalen)
(3 rows)

